agreementbot/agreementbot.go

```diff
@@ -797,14 +797,22 @@ func DeleteMessage(msgId int, agbotId, agbotToken, exchangeURL string, httpClien
 // microservices. If the agbot is working with a policy file that was generated from a pattern, then it will do searches
 // by pattern. If the agbot is working with a manually created policy file, then it will do searches by list of
 // microservices.
-func (w *AgreementBotWorker) searchExchange(pol *policy.Policy, searchOrg string) (*[]exchange.SearchResultDevice, error) {
+func (w *AgreementBotWorker) searchExchange(pol *policy.Policy, polOrg string) (*[]exchange.SearchResultDevice, error) {
 
 	// If it is a pattern based policy, search by worload URL and pattern.
 	if pol.PatternId != "" {
+		// get a list of node orgs that agbot is serving for this pattern
+		nodeOrgs := w.PatternManager.GetServedNodeOrgs(polOrg, exchange.GetId(pol.PatternId))
+		if len(nodeOrgs) == 0 {
+			glog.V(3).Infof("Policy file for pattern %v exits but currently the agbot is not serving this policy for any organizations.", pol.PatternId)
```

Comment:
typo: "pattern %v exits", should be "exists"

Reply:
fixed.